renderer/components/js/components.js

```diff
@@ -0,0 +1,123 @@
+org.ekstep.questionunit = org.ekstep.questionunit || {};
+org.ekstep.questionunit.baseComponent = {
+    playAudio: function (audioObj) {
+        EkstepRendererAPI.dispatchEvent('org.ekstep.questionunit:playaudio', audioObj)
+    },
+    loadImageFromUrl: function (element, imgUrl, pluginId, pluginVer) {
+        EkstepRendererAPI.dispatchEvent('org.ekstep.questionunit:loadimagefromurl', { 'element': element, 'path': imgUrl, 'pluginId': pluginId, 'pluginVer': pluginVer });
+    },
+    loadAssetUrl: function(element, imgUrl, pluginId, pluginVer){
+        EkstepRendererAPI.dispatchEvent('org.ekstep.questionunit:loadAssetUrl', { 'element': element, 'path': imgUrl, 'pluginId': pluginId, 'pluginVer': pluginVer });
+    },
+    generateModelTemplate: function () {
+        return "<div class='popup' id='image-model-popup' onclick='org.ekstep.questionunit.questionComponent.hideImageModel()'><div class='popup-overlay' onclick='org.ekstep.questionunit.questionComponent.hideImageModel()'></div> \
+        <div class='popup-full-body'> \
+            <div class='font-lato assess-popup assess-goodjob-popup'> \
+                <img class='qc-question-fullimage' src=<%= src %> /> \
+                <div onclick='org.ekstep.questionunit.questionComponent.hideImageModel()' class='qc-popup-close-button'>&times;</div> \
+            </div>\
+        </div>"
+    },
+    showImageModel: function (event, imageSrc, elementId) {
```

Comment:
Make this method as generic taking object as an input
{ 
  target: // actual element clicked
  src: // If user want to pass source
  template: // If user want to show template popup
}

Reply:
Think of it more generic, where all question images will be read & onclick event is binded for those to show modal popup.